### ISSUE TITLE
question please

### ISSUE THREAD

**Issue by ahmedabbas81** [shap/shap#990]
@farh33n @slundberg
My neural network has two hidden layers and one output layer. The hidden layers have relu activation function and the output layer is sigmoid and it is multi-class classification NN.

Is using deepshap with that neural network model correct?

I only use
e = shap.DeepExplainer(model, background)
shap_values = e.shap_values(x_test[1:5])

as in the example in your github page. Is that correct?
Do I need to perform any processing on the contribution values that I get (I mean shap_values)? Or I can rely on them directly as the contribution values that I want.

Also, when I use softmax instead of sigmoid, I get other contribution values that are less reasonable than those I get when using sigmoid (there are important inputs that contribute poorly when using softmax). Is there a reason for that?

And a general question, can I use sigmoid as the activation function in a multi-class classification problem where the classes are mutually exclusive, and take the maximum output as the correct class?

Thanks

**Comment by amjass12** [comment]
I have used SHAP to also work on feature contirbutions using a NN with a final sigmoid output. The reason being the sigmoid output allows for samples to belong to multiple classes:

Do I need to perform any processing on the contribution values that I get (I mean shap_values)? Or I can rely on them directly as the contribution values that I want.

The shap_values should represent the contribution values you are looking for. 

And a general question, can I use sigmoid as the activation function in a multi-class classification problem where the classes are mutually exclusive, and take the maximum output as the correct class?

If the classes are mutually exclusive you will want to use a softmax....

other people can correct me if any of this is wrong!

**Comment by ahmedabbas81** [comment]
@amjass12  Thank you for your answer.

I know that deepSHAP uses several instances of random reference inputs. For example, it uses 100 instances of random inputs as reference (background inputs), and calculates the average contribution over these 100 instances.

My question is: is it wrong to use only one reference input of zeros (this is the suitable reference to my project)?

The second question is: in my case (a neural network with two hidden layers only of relu activation and an output layer), is there any difference between deepSHAP and deepLIFT?

I am asking this second question because I learnt that deepSHAP is for tree classifiers.

Thank you

**Comment by slundberg** [comment]
@ahmedabbas81 

1. It is fine to use a single background reference, it just means the your prior background distribution is a point function at zero.

2. Deep SHAP is a way to combine SHAP values for smaller components into a larger model by assuming linear combination rules (hence it is an approximation for non-linear NNs). It is not for trees, though it can be used to combine SHAP values from trees with later transformations or model compositions. DeepLIFT rescale is the same as Deep SHAP for relu only models. If you have cases where it is not working well I would compare to GradientExplainer and then post an example if possible.
